packages/skeleton-react/src/lib/components/Segment/Segment.tsx

```diff
@@ -1,113 +1,102 @@
 });
 
 // Components ---
 
 const SegmentRoot: React.FC<SegmentProps> = ({
-	value = '',
-	name = '',
 	// Root
 	base = 'inline-flex items-stretch overflow-hidden',
 	background = 'preset-outlined-surface-200-800',
 	border = 'p-2',
-	flexDirection = 'flex-row', // vertical: flex-col
 	gap = 'gap-2',
 	padding = '',
 	rounded = 'rounded-container',
 	width = '',
 	classes = '',
+	// Indicator
+	indicatorBase = 'top-[var(--top)] left-[var(--left)] w-[var(--width)] h-[var(--height)]',
+	indicatorBg = 'preset-filled',
+	indicatorRounded = 'rounded',
+	indicatorClasses = '',
 	// Events
-	onChange,
+	onValueChange = noop,
 	// Children
-	children
+	children,
+	// Zag
+	...zagProps
 }) => {
-	function onSelectionHandler(newValue: string) {
-		value = newValue;
-		if (onChange) onChange(newValue);
-	}
+	const [state, send] = useMachine(
+		radio.machine({
+			id: useId(),
+			onValueChange: (details) => {
+				onValueChange(details.value);
+			}
+		}),
+		{ context: zagProps }
+	);
+	const api = radio.connect(state, send, normalizeProps);
 
 	// Set Context
-	const ctx = {
-		value,
-		name,
-		onSelectionHandler
-	};
+	const ctx = { api };
+
+	// Reactive
+	// FIXME: always returns vertical
+	// const rxOrientation = state.context.orientation === 'vertical' ? 'flex-col' : 'flex-row';
+	const rxOrientation = 'flex-row';
 
 	return (
 		<div
-			role="radiogroup"
-			className={`${base} ${flexDirection} ${background} ${border} ${padding} ${gap} ${rounded} ${width} ${classes}`}
+			{...api.getRootProps()}
+			className={`${base} ${rxOrientation} ${background} ${border} ${padding} ${gap} ${rounded} ${width} ${classes}`}
 			data-testid="segment"
 		>
+			{/* Indicator */}
+			<div {...api.getIndicatorProps()} className={`${indicatorBase} ${indicatorBg} ${indicatorRounded} ${indicatorClasses}`}></div>
+			{/* Items */}
 			<SegmentContext.Provider value={ctx}>{children}</SegmentContext.Provider>
 		</div>
 	);
 };
 
 const SegmentItem: React.FC<SegmentItemsProps> = ({
-	id,
 	value,
-	title,
-	disabled = false,
 	// Root
-	base = 'btn',
-	active = 'preset-filled',
-	hover = 'hover:preset-tonal',
+	base = 'btn cursor-pointer z-[1]',
 	classes = '',
 	// Label
 	labelBase = 'pointer-events-none',
+	labelActiveText = 'preset-filled',
 	labelClasses = '',
-	// Events
-	onClick,
 	// Children
 	children
 }) => {
 	// Get Context
-	const ctx = useContext<SegmentContextState>(SegmentContext);
+	const ctx = useContext(SegmentContext);
 
 	// Reactive
-	const [selected, setSelected] = useState(value === ctx.value);
-	const [rxActive, setRxActive] = useState(selected ? active : hover);
-
-	useEffect(() => {
-		setSelected(value === ctx.value);
-	}, [value, ctx.value]);
-
-	useEffect(() => {
-		setRxActive(selected ? active : hover);
-	}, [selected, active, hover]);
-
-	function onClickHandler() {
-		ctx.onSelectionHandler(value);
-		onClick?.(value);
-	}
+	const rxActive = ctx.api.value === value;
+	const rxActiveText = rxActive ? labelActiveText : '';
```

Comment:
You can do:
```suggestion
	// Reactive
    const state = ctx.api.getItemState(zagProps);  
    const activeClasses = state.active ? labelActiveText : ''; 
```

Reply:
Per Discord, this introduces type errors when trying to use `zagProps`. I tried a few options but no luck. Would welcome your suggestions for how to resolve this.